### DISCUSSION TITLE
ImportError: cannot import name 'VERSION' from 'serial' (unknown location)

### DISCUSSION THREAD

**Discussion by dichaos** [eblot/pyftdi#235] [Q&A] [answered]
hello,

I installed the latest version of pyftdi via pip. I upgrades all of my packages to the latest version too but when I try to load pyftdi.serialext I get the error message

Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/pyftdi/serialext/__init__.py", line 15, in <module>
    from serial import VERSION, serial_for_url as serial4url
ImportError: cannot import name 'VERSION' from 'serial' (unknown location)

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/dikaios.papadogkonas/Desktop/lidar.py", line 7, in <module>
    import pyftdi.serialext
  File "/usr/local/lib/python3.9/site-packages/pyftdi/serialext/__init__.py", line 20, in <module>
    raise ImportError("pyserial 3.0+ is required") from exc
ImportError: pyserial 3.0+ is required

I have installed pyserial 3.5 

Any idea why it can't find VERSION ?

**Comment by eblot** [comment]
Weird.

`serial.VERSION` is define here:

https://github.com/pyserial/pyserial/blob/0e7634747568547b8a7f9fd0c48ed74f16af4b23/serial/__init__.py#L20

Maybe try to use Python REPL and check which file is actually imported:
```
>>> import serial
>>> print serial.__file__
```
and check the actual content of this file.

**Reply by dichaos** [reply]
Thank you for your reply, interesting, I get "None"

**Comment by eblot** [comment, marked as answer]
Should be something like:
```
> python3
Python 3.8.6 (default, Nov 21 2020, 23:20:31)
[Clang 12.0.0 (clang-1200.0.32.27)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import serial
>>> serial.VERSION
'3.5'
>>> serial.__file__
'/usr/local/lib/python3.8/site-packages/serial/__init__.py'
```

Try uninstalling `pyserial` and check that `import serial` fails with an `ImportError`.

Bw sure you do no have a `serial.py` somewhere in your `PYTHONPATH` that could be imported instead of pyserial's.

**Reply by dichaos** [reply]
Actually uninstalling and re-installing fixed it. Sorry about that and thank you
